### DISCUSSION TITLE
Restore checkpoint from pretrained model

### DISCUSSION THREAD

**Discussion by JRMeyer** [coqui-ai/TTS#273] [Archived]
<USER>>>> CrazyJoeDevola<br/> <DATE> [September 15, 2020, 3:03pm]<br/> <br/> Hi<br/> <br/> Not sure if and what i am doing wrong, but i downloaded one of the LJ<br/> trained models of 130k. I restored training after adding some missing<br/> parameters. I would expect the model to produce audible voice after 10<br/> epochs or so, due to the pretraining (i suppose this is the whole<br/> purpose). but Even now, at step 165725, there is only echo in the audio<br/> previews.<br/> <br/> Is this due to the fact that i cloned the latest DEV repository, and<br/> should have trained based on a different, specific branch?<br/> <br/> I was looking for a shortcut to do a custom voice, with the basics of<br/> the language already learnt, and thereby improving the results of the<br/> custom voice.<br/> <br/> [This is an archived TTS discussion thread from discourse.mozilla.org/t/restore-checkpoint-from-pretrained-model]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> erogol<br/> <DATE> [September 15, 2020, 3:08pm]<br/> <br/> it is better if you use the same commit as the checkpoint version.<br/> <br/> If it is something about your training, we need more info to help.<br/> Tensorboard output, examples audios etc.<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> CrazyJoeDevola<br/> <DATE> [September 15, 2020, 4:24pm]<br/> <br/> Test audios:<br/> <https://drive.google.com/drive/folders/1D2vIqx2oQzKXPSVa1dyLnKgtAN2rjiJu?usp=sharing><br/> <br/> config<br/> <br/> > { *slash* <br/> > 'model': 'Tacotron2', *slash* <br/> > 'run_name': 'ljspeech-tac2', *slash* <br/> > 'run_description': 'tacotron2 mean-scale normalization with gradual<br/> > training',<br/> ><br/> > // AUDIO PARAMETERS<br/> > 'audio':{<br/> > // stft parameters<br/> > 'fft_size': 1024, // number of stft frequency levels. Size of the linear spectogram frame.<br/> > 'win_length': 1024, // stft window length in ms.<br/> > 'hop_length': 256, // stft window hop-lengh in ms.<br/> > 'frame_length_ms': null, // stft window length in ms.If null, 'win_length' is used.<br/> > 'frame_shift_ms': null, // stft window hop-lengh in ms. If null, 'hop_length' is used.<br/> ><br/> > // Audio processing parameters<br/> > 'sample_rate': 22050, // DATASET-RELATED: wav sample-rate. If different than the original data, it is resampled.<br/> > 'preemphasis': 0.0, // pre-emphasis to reduce spec noise and make it more structured. If 0.0, no -pre-emphasis.<br/> > 'ref_level_db': 0, // reference level db, theoretically 20db is the sound of air.<br/> ><br/> > // Silence trimming<br/> > 'do_trim_silence': true,// enable trimming of slience of audio as you load it. LJspeech (false), TWEB (false), Nancy (true)<br/> > 'trim_db': 60, // threshold for timming silence. Set this according to your dataset.<br/> ><br/> > // Griffin-Lim<br/> > 'power': 1.5, // value to sharpen wav signals after GL algorithm.<br/> > 'griffin_lim_iters': 60,// #griffin-lim iterations. 30-60 is a good range. Larger the value, slower the generation.<br/> ><br/> > // MelSpectrogram parameters<br/> > 'num_mels': 80, // size of the mel spec frame.<br/> > 'mel_fmin': 50.0, // minimum freq level for mel-spec. ~50 for male and ~95 for female voices. Tune for dataset!!<br/> > 'mel_fmax': 7600.0, // maximum freq level for mel-spec. Tune for dataset!!<br/> > 'spec_gain': 1.0, // scaler value appplied after log transform of spectrogram.<br/> ><br/> > // Normalization parameters<br/> > 'signal_norm': true, // normalize spec values. Mean-Var normalization if 'stats_path' is defined otherwise range normalization defined by the other params.<br/> > 'min_level_db': -100, // lower bound for normalization<br/> > 'symmetric_norm': true, // move normalization to range [-1, 1]<br/> > 'max_norm': 4.0, // scale normalization to range [-max_norm, max_norm] or [0, max_norm]<br/> > 'clip_norm': true, // clip normalized values into the range.<br/> > 'stats_path': null // DO NOT USE WITH MULTI_SPEAKER MODEL. scaler stats file computed by 'compute_statistics.py'. If it is defined, mean-std based notmalization is used and other normalization params are ignored<br/> > },<br/> ><br/> > // VOCABULARY PARAMETERS<br/> > // if custom character set is not defined,<br/> > // default set in symbols.py is used<br/> > // 'characters':{<br/> > // 'pad': '_',<br/> > // 'eos': '~',<br/> > // 'bos': '^',<br/> > // 'characters': 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz!'(),-.:;? ',<br/> > // 'punctuations':'!'(),-.:;? ',<br/> > // 'phonemes':'iyɨʉɯuɪʏʊeøɘəɵɤoɛœɜɞʌɔæɐaɶɑɒᵻʘɓǀɗǃʄǂɠǁʛpbtdʈɖcɟkɡqɢʔɴŋɲɳnɱmʙrʀⱱɾɽɸβfvθðszʃʒʂʐçʝxɣχʁħʕhɦɬɮʋɹɻjɰlɭʎʟˈˌːˑʍwɥʜʢʡɕʑɺɧɚ˞ɫ'<br/> > // },<br/> ><br/> > // DISTRIBUTED TRAINING<br/> > 'distributed':{<br/> > 'backend': 'nccl',<br/> > 'url': 'tcp: *slash* / *slash* /localhost:54321'<br/> > },<br/> ><br/> > 'reinit_layers': [], // give a list of layer names to restore from the given checkpoint. If not defined, it reloads all heuristically matching layers.<br/> ><br/> > // TRAINING<br/> > 'batch_size': 32, // Batch size for training. Lower values than 32 might cause hard to learn attention. It is overwritten by 'gradual_training'.<br/> > 'eval_batch_size':16,<br/> > 'r': 7, // Number of decoder frames to predict per iteration. Set the initial values if gradual training is enabled.<br/> > 'gradual_training': [[0, 7, 64], [1, 5, 64], [50000, 3, 32], [130000, 2, 32], [290000, 1, 32]], //set gradual training steps [first_step, r, batch_size]. If it is null, gradual training is disabled. For Tacotron, you might need to reduce the 'batch_size' as you proceeed.<br/> > 'loss_masking': true, // enable / disable loss masking against the sequence padding.<br/> > 'ga_alpha': 1.0, // weight for guided attention loss. If > 0, guided attention is enabled.<br/> > 'apex_amp_level': null, // level of optimization with NVIDIA's apex feature for automatic mixed FP16/FP32 precision (AMP), NOTE: currently only O1 is supported, and use 'O1' to activate.<br/> ><br/> ><br/> > // VALIDATION<br/> > 'run_eval': true,<br/> > 'test_delay_epochs': 10, //Until attention is aligned, testing only wastes computation time.<br/> > 'test_sentences_file': null, // set a file to load sentences to be used for testing. If it is null then we use default english sentences.<br/> ><br/> > // OPTIMIZER<br/> > 'noam_schedule': false, // use noam warmup and lr schedule.<br/> > 'grad_clip': 1.0, // upper limit for gradients for clipping.<br/> > 'epochs': 1000, // total number of epochs to train.<br/> > 'lr': 0.0003, // Initial learning rate. If Noam decay is active, maximum learning rate.<br/> > 'wd': 0.000001, // Weight decay weight.<br/> > 'warmup_steps': 4000, // Noam decay steps to increase the learning rate from 0 to 'lr'<br/> > 'seq_len_norm': false, // Normalize eash sample loss with its length to alleviate imbalanced datasets. Use it if your dataset is small or has skewed distribution of sequence lengths.<br/> ><br/> > // TACOTRON PRENET<br/> > 'memory_size': -1, // ONLY TACOTRON - size of the memory queue used fro storing last decoder predictions for auto-regression. If < 0, memory queue is disabled and decoder only uses the last prediction frame.<br/> > 'prenet_type': 'original', // 'original' or 'bn'.<br/> > 'prenet_dropout': true, // enable/disable dropout at prenet.<br/> ><br/> > // ATTENTION<br/> > 'attention_type': 'original', // 'original' or 'graves'<br/> > 'attention_heads': 4, // number of attention heads (only for 'graves')<br/> > 'attention_norm': 'sigmoid', // softmax or sigmoid. Suggested to use softmax for Tacotron2 and sigmoid for Tacotron.<br/> > 'windowing': false, // Enables attention windowing. Used only in eval mode.<br/> > 'use_forward_attn': true, // if it uses forward attention. In general, it aligns faster.<br/> > 'forward_attn_mask': false, // Additional masking forcing monotonicity only in eval mode.<br/> > 'transition_agent': false, // enable/disable transition agent of forward attention.<br/> > 'location_attn': true, // enable_disable location sensitive attention. It is enabled for TACOTRON by default.<br/> > 'bidirectional_decoder': false, // use https://arxiv.org/abs/1907.09006. Use it, if attention does not work well with your dataset.<br/> > 'double_decoder_consistency': false, // use DDC explained here https://erogol.com/solving-attention-problems-of-tts-models-with-double-decoder-consistency-draft/<br/> > 'ddc_r': 1, // reduction rate for coarse decoder.<br/> ><br/> > // STOPNET<br/> > 'stopnet': true, // Train stopnet predicting the end of synthesis.<br/> > 'separate_stopnet': true, // Train stopnet seperately if 'stopnet==true'. It prevents stopnet loss to influence the rest of the model. It causes a better model, but it trains SLOWER.<br/> ><br/> > // TENSORBOARD and LOGGING<br/> > 'print_step': 25, // Number of steps to log training on console.<br/> > 'tb_plot_step': 100, // Number of steps to plot TB training figures.<br/> > 'print_eval': false, // If True, it prints intermediate loss values in evalulation.<br/> > 'save_step': 10000, // Number of training steps expected to save traninpg stats and checkpoints.<br/> > 'checkpoint': true, // If true, it saves checkpoints per 'save_step'<br/> > 'tb_model_param_stats': false, // true, plots param stats per layer on tensorboard. Might be memory consuming, but good for debugging.<br/> ><br/> > // DATA LOADING<br/> > 'text_cleaner': 'phoneme_cleaners',<br/> > 'enable_eos_bos_chars': false, // enable/disable beginning of sentence and end of sentence chars.<br/> > 'num_loader_workers': 4, // number of training data loader processes. Don't set it too big. 4-8 are good values.<br/> > 'num_val_loader_workers': 4, // number of evaluation data loader processes.<br/> > 'batch_group_size': 4, //Number of batches to shuffle after bucketing.<br/> > 'min_seq_len': 6, // DATASET-RELATED: minimum text length to use in training<br/> > 'max_seq_len': 153, // DATASET-RELATED: maximum text length<br/> ><br/> > // PATHS<br/> > 'output_path': '/content/drive/My Drive/Pretrained2.0',<br/> ><br/> > // PHONEMES<br/> > 'phoneme_cache_path': '/content/drive/My Drive/Pretrained2.0/cache', // phoneme computation is slow, therefore, it caches results in the given folder.<br/> > 'use_phonemes': true, // use phonemes instead of raw characters. It is suggested for better pronounciation.<br/> > 'phoneme_language': 'en-us', // depending on your target language, pick one from https://github.com/bootphon/phonemizer#languages<br/> ><br/> > // MULTI-SPEAKER and GST<br/> > 'use_speaker_embedding': false, // use speaker embedding to enable multi-speaker learning.<br/> > 'use_external_speaker_embedding_file': false, // if true, forces the model to use external embedding per sample instead of nn.embeddings, that is, it supports external embeddings such as those used at: https://arxiv.org/abs /1806.04558<br/> > 'external_speaker_embedding_file': '../../speakers-vctk-en.json', // if not null and use_external_speaker_embedding_file is true, it is used to load a specific embedding file and thus uses these embeddings instead of nn.embeddings, that is, it supports external embeddings such as those used at: https://arxiv.org/abs /1806.04558<br/> > 'use_gst': false, // use global style tokens<br/> > 'gst': { // gst parameter if gst is enabled<br/> > 'gst_style_input': null, // Condition the style input either on a<br/> > // -> wave file [path to wave] or<br/> > // -> dictionary using the style tokens {'token1': 'value', 'token2': 'value'} example {'0': 0.15, '1': 0.15, '5': -0.15}<br/> > // with the dictionary being len(dict) <= len(gst_style_tokens).<br/> > 'gst_embedding_dim': 512,<br/> > 'gst_num_heads': 4,<br/> > 'gst_style_tokens': 10<br/> ><br/> > }, *slash* <br/> > // DATASETS *slash* <br/> > 'datasets': // List of datasets. They all merged and they get<br/> > different speaker_ids. *slash* <br/> > *slash* [ *slash* <br/> > { *slash* <br/> > 'name': 'ljspeech', *slash* <br/> > 'path': '/content/LJSpeech-1.1', *slash* <br/> > 'meta_file_train': 'metadata.csv', *slash* <br/> > 'meta_file_val': 'metadata_val.csv' *slash* <br/> > } *slash* <br/> > *slash* ]<br/> ><br/> > }<br/> <br/> > EVALUATION<br/> <br/> *slash* | *slash* > Synthesizing test sentences *slash* <br/> *slash* | *slash* > Decoder stopped with *slash* 'max_decoder_steps *slash* <br/> *slash* | *slash* > Decoder stopped with *slash* 'max_decoder_steps *slash* <br/> *slash* | *slash* > Decoder stopped with *slash* 'max_decoder_steps *slash* <br/> warning: audio amplitude out of range, auto clipped. *slash* <br/> *slash* | *slash* > avg_decoder_loss: 0.00337 (-0.00013) *slash* <br/> *slash* | *slash* > avg_postnet_loss: 0.00334 (-0.00008) *slash* <br/> *slash* | *slash* > avg_stopnet_loss: 0.15691 (-0.00884) *slash* <br/> *slash* | *slash* > avg_ga_loss: 0.00073 (-0.00026) *slash* <br/> *slash* | *slash* > avg_loss: 0.00744 (-0.00047) *slash* <br/> *slash* | *slash* > avg_align_error: 0.76313 (-0.01047)<br/> <br/> [Archived Post]<br/>
